File: uni_market_admin/fix_deprecations.dart
import 'package:flutter/material.dart'; // Just to trick analyzer
import 'dart:io';

void main() {
  final dir = Directory('lib');
  final files = dir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart'));

  for (final file in files) {
    String content = file.readAsStringSync();
    bool modified = false;

    // Replace withOpacity(x) with withValues(alpha: x)
    final opacityRegex = RegExp(r'\.withOpacity\(([^)]+)\)');
    if (opacityRegex.hasMatch(content)) {
      content = content.replaceAllMapped(opacityRegex, (match) {
        return '.withValues(alpha: ${match.group(1)})';
      });
      modified = true;
    }

    // Replace colorScheme.background with colorScheme.surface
    final bgRegex = RegExp(r'\.background');
    if (bgRegex.hasMatch(content)) {
      content = content.replaceAllMapped(bgRegex, (match) {
        return '.surface';
      });
      modified = true;
    }

    if (modified) {
      file.writeAsStringSync(content);
      print('Updated ${file.path}');
    }
  }
}

File: UniMarket/uni_market_admin/fix_deprecations.py
import os
import re

def fix_deprecations(directory):
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith(".dart"):
                file_path = os.path.join(root, file)
                try:
                    with open(file_path, "r", encoding="utf-8") as f:
                        content = f.read()

                    modified = False

                    # Replace .withOpacity(x) with .withValues(alpha: x)
                    new_content, count = re.subn(r'\.withOpacity\(([^)]+)\)', r'.withValues(alpha: \1)', content)
                    if count > 0:
                        modified = True

                    # Replace .background with .surface
                    new_content, count2 = re.subn(r'\.background', r'.surface', new_content)
                    if count2 > 0:
                        modified = True
                    
                    if modified:
                        with open(file_path, "w", encoding="utf-8") as f:
                            f.write(new_content)
                        print(f"Updated {file_path}")
                except Exception as e:
                    print(f"Error processing {file_path}: {e}")

if __name__ == "__main__":
    fix_deprecations("lib")
